### PR TITLE
Support parsing optional nulls handling for unique constraint

### PR DESCRIPTION
**Pull request by mvzink**

### CODE REVIEW

File: tests/sqlparser_postgres.rs [hunk 1]
```diff
@@ -594,6 +594,13 @@ fn parse_alter_table_constraints_rename() {
     }
 }
 
+#[test]
+fn parse_alter_table_constraints_unique_nulls_distinct() {
+    pg_and_generic().verified_stmt("ALTER TABLE t ADD CONSTRAINT b UNIQUE NULLS NOT DISTINCT (c)");
```
Comment (iffyio) [inline tests/sqlparser_postgres.rs:599]: For one of the test cases can we include an assertion on the AST, like in [this style](https://github.com/apache/datafusion-sqlparser-rs/blob/a54331889874ca896100b6666720d6fcb43b6965/tests/sqlparser_postgres.rs#L882-L888)? just to sanity check where in the statement the nulls option ends up in the tree

**Review by iffyio — COMMENTED** [review]
Thanks @mvzink! Looks good to me just a minor comment on the test

**Review by iffyio — APPROVED** [review]
LGTM! Thanks @mvzink!
cc @alamb

**Comment by alamb** [comment]
Looks good to me -- feel free to merge the PRs in yourself @iffyio  as well -- I appreciate the heads up, but you should totally feel empowered to merge directly yourself too

**Pull request merged**